pkg/madmin/api.go

```diff
@@ -381,6 +383,20 @@ func (adm AdminClient) executeMethod(ctx context.Context, method string, reqData
 			}
 		}
 
+		// Certain operations might have been completed
+		// such as
+		// - RemoveUser
+		// - RemoveCannedPolicy
+		// - UpdateGroupMembers
+		//
+		// Due to a potential network timeout, if we are in
+		// a retry situation we should ignore if there is
+		// http.StatusNotFound error returned.
+		if retry && res.StatusCode == http.StatusNotFound {
```

Comment:
This is a too low-level check, it doesn't not enough context.

Besides the client would notice if there is a network timeout and should show it.

I guess what we can do is return success if we want to remove a non existent user.

Reply:
The client won't notice @vadmeste we have auto-retry nothing is visible so if there was a network error - this is to ensure that we don't fail unnecessarily upon 404. The logic is that this happens only during a `retry` 

> I guess what we can do is return success if we want to remove a non existent user.

I already did that @vadmeste